subprojects/ide/src/main/groovy/org/gradle/plugins/ide/eclipse/EclipseWtpPlugin.groovy

```diff
@@ -67,53 +69,60 @@ class EclipseWtpPlugin extends IdePlugin {
     }
 
     private void configureEclipseClasspath(Project project, EclipseModel eclipseModel) {
+
+        def adjustClasspath = { Classpath classpath ->
+            def libDeployPath = eclipseModel.wtp.component.libDeployPath
+            def depFiles = WtpComponentFactory.resolveDependenciesFor(eclipseModel.wtp.component.plusConfigurations,
+                    eclipseModel.wtp.component.minusConfigurations)*.file
+
+            for (entry in classpath.entries) {
+                if (!(entry instanceof AbstractLibrary)) {
+                    continue;
+                }
+                if (depFiles.contains(entry.library.file)) {
+                    // '../' and '/WEB-INF/lib' both seem to be correct (and equivalent) values here
+                    //this is necessary so that the depended upon projects will have their dependencies
+                    // deployed to WEB-INF/lib of the main project.
+                    entry.entryAttributes[AbstractClasspathEntry.COMPONENT_DEPENDENCY_ATTRIBUTE] = libDeployPath
+                } else {
+                    // Mark this library as not required for deployment
+                    entry.entryAttributes[AbstractClasspathEntry.COMPONENT_NON_DEPENDENCY_ATTRIBUTE] = ''
+                }
+            }
+        }
+
+        def adjustWtpComponent = { WtpComponent wtpComponent ->
+            def depPaths = ClasspathFactory.resolveDependenciesFrom(eclipseModel.classpath).collect { dependency ->
+                eclipseModel.classpath.fileReferenceFactory.fromFile(dependency.file).path
+            }
+            wtpComponent.wbModuleEntries.removeAll{ wbModule ->
+                wbModule instanceof WbDependentModule && depPaths.any{ path -> wbModule.handle.endsWith(path) }
+            }
+        }
+
         project.plugins.withType(JavaPlugin) {
-            def deleteWtpDependentModule = []
             eclipseModel.classpath.file.whenMerged { Classpath classpath ->
                 if (hasWarOrEarPlugin(project)) {
```

Comment:
Hi @mark-vieira, thanks for the Information. Does this mean the default settings of a WAR+EAR project should be from the WAR project and the EAR stuff has to be setup manually? Do you have an example project or build.gradle for me to write an automatic test for this?